src/bindings/c/include/c_api/ov_c_api.h

```diff
@@ -0,0 +1,1390 @@
+                      ov_compiled_model_t** compiled_model,
+                      const ov_property_t* property);
+
+/**
+ * @brief Reads a model and creates a compiled model from the IR/ONNX/PDPD file.
+ * This can be more efficient than using the ov_core_read_model_from_XXX + ov_core_compile_model flow,
+ * especially for cases when caching is enabled and a cached model is available.
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_path Path to a model.
+ * @param device_name Name of a device to load a model to.
+ * @param compiled_model A pointer to the newly created compiled_model.
+ * @param property Optional pack of pairs: (property name, property value) relevant only for this load operation
+ * operation.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_compile_model_from_file(const ov_core_t* core,
+                                const char* model_path,
+                                const char* device_name,
+                                ov_compiled_model_t** compiled_model,
+                                const ov_property_t* property);
+
+/**
+ * @brief Sets properties for a device, acceptable keys can be found in ov_property_key_e.
+ * @param core A pointer to the ie_core_t instance.
+ * @param device_name Name of a device.
+ * @param property ov_property propertys.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_set_property(const ov_core_t* core, const char* device_name, const ov_property_t* property);
+
+/**
+ * @brief Gets properties related to device behaviour.
+ * The method extracts information that can be set via the set_property method.
+ * @param core A pointer to the ie_core_t instance.
+ * @param device_name  Name of a device to get a property value.
+ * @param property_name  Property name.
+ * @param property_value A pointer to property value.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_get_property(const ov_core_t* core,
+                     const char* device_name,
+                     const ov_property_key_e property_name,
+                     ov_property_value_t* property_value);
+
+/**
+ * @brief Returns devices available for inference.
+ * @param core A pointer to the ie_core_t instance.
+ * @param devices A pointer to the ov_available_devices_t instance.
+ * Core objects go over all registered plugins and ask about available devices.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e) ov_core_get_available_devices(const ov_core_t* core, ov_available_devices_t* devices);
+
+/**
+ * @brief Releases memory occpuied by ov_available_devices_t
+ * @param devices A pointer to the ov_available_devices_t instance.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(void) ov_available_devices_free(ov_available_devices_t* devices);
```

Comment:
You don't have status code in this function.
BTW how do you report an error if memory cannot be deallocated?

Reply:
For C APIs, I don't think it is a must requirement, maybe we cannot pick the bone from the egg, right?